### DISCUSSION TITLE
bulk_load against existing triples

### DISCUSSION THREAD

**Discussion by ch-sander** [oxigraph/oxigraph#1090] [Q&A]
I'm using pyoxigraph's `Store.bulk_load`. If the Store is already populated, I don't manage to change existing triples via bulk loading (from TTL to named graph). Is this intended and requires the `bulk_extend` function (which however doesn't load from TTL directly but requires quads)?

**Comment by Tpt** [comment]
`bulk_load` adds new quads to the store. If you want to replace the store content use first `Store.clear()` to remove the store content (or, even simpler and faster, remove the store entirely and create a new one)

**Comment by ch-sander** [comment]
That' what I'm doing (and it works). I was instead wondering if I could update only "new" triples, but this doesn't seem to work this way.

> remove the store entirely
you mean, delete the folder/files? This is only an OS operation or also a trigger in pyoxigraph as well?

**Reply by Tpt** [reply]
>  you mean, delete the folder/files? This is only an OS operation or also a trigger in pyoxigraph as well?

Indeed, this is only an OS operation.
